### PR TITLE
Fix failed e2e tests for dns configmap

### LINKED ISSUE #62490: DNS configMap tests are failing in serial CI
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind e2e-test-failure 
/priority failing-test 
/sig network

**What happened**:
From http://k8s-testgrid.appspot.com/sig-network-gce#gci-gce-serial&width=20,  DNS configMap tests are failing since https://github.com/kubernetes/kubernetes/pull/62042 was merged with below error:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/dns_configmap.go:38
You may only call BeforeEach from within a Describe or Context
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:141
```

@grayluck

### PR DESCRIPTION
**Pull request by grayluck**
**What this PR does / why we need it**:
Fix DNS configMap e2e tests.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #62490 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by grayluck** [comment]
\assign @MrHohn

File: test/e2e/network/dns_configmap.go [hunk 1]
```diff
@@ -220,14 +221,15 @@ func (t *dnsPtrFwdTest) run() {
 
 var _ = SIGDescribe("DNS configMap nameserver", func() {
 
+	nsTest := &dnsNameserverTest{dnsTestCommon: newDnsTestCommon()}
+	fwdTest := &dnsPtrFwdTest{dnsTestCommon: newDnsTestCommon()}
```
Comment (MrHohn) [inline test/e2e/network/dns_configmap.go]: Interesting, didn't notice `newDnsTestCommon()` is embedded with `NewDefaultFramework()`, which is embedded with `BeforeEach` and `AfterEach`.

By calling `newDnsTestCommon()` twice, we push two BeforeEach/AfterEach hooks, which also seems improper. Can we move `framework.NewDefaultFramework()` out of `newDnsTestCommon` so we don't call it twice?

Comment (grayluck) [reply test/e2e/network/dns_configmap.go]: Wrap two tests into different contexts. This should do the trick. (*msg will be squashed

**Review by MrHohn — APPROVED** [review]
Oh this looks nice! Have you tried it for real? :)

/lgtm

**Comment by grayluck** [comment]
I tried of course. Forgot to compile before running the test scripts last time.
Thanks!

**Pull request merged**